### PR TITLE
Look for an existing jspecify jar file and build the project if needed

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

File: demo [hunk 1]
```diff
@@ -8,12 +8,20 @@ jspecify="${dir}/../jspecify/build/libs/jspecify-0.0.0-SNAPSHOT.jar"
...
+    mvn -q org.apache.maven.plugins:maven-dependency-plugin:3.6.1:copy \
+      -Dartifact="org.jspecify:jspecify:${version}" \
+      -DoutputDirectory="$(dirname "${jspecify}")"
+  fi
 fi
-ourclasspath="${jspecify}:${dir}/build/libs/jspecify-reference-checker.jar"
+jspecifyReferenceChecker="${dir}/build/libs/jspecify-reference-checker.jar"
+if [ ! -e "${jspecifyReferenceChecker}" ]; then
+  echo "Assembling jspecify-reference-checker"
+  ./gradlew assemble
```
Comment (wmdietl) [inline demo:21]: This should also ensure that the `$dir/../checker-framework/checker/bin/javac` below exists, so somebody should just be able to clone this repo and immediately run `./demo`.
Alternatively, we could give an error message here.

Comment (netdpb) [reply demo:21]: This is a good idea. Can you also update the README to no longer say you have to run `./gradlew assemble`?

File: demo [hunk 2]
```diff
@@ -8,12 +8,20 @@ jspecify="${dir}/../jspecify/build/libs/jspecify-0.0.0-SNAPSHOT.jar"
...
-    -DoutputDirectory="$(dirname "${jspecify}")"
+  if [ ! -e "${jspecify}" ]; then
+    echo "Downloading $(basename "${jspecify}") from Maven central"
+    mvn -q org.apache.maven.plugins:maven-dependency-plugin:3.6.1:copy \
+      -Dartifact="org.jspecify:jspecify:${version}" \
+      -DoutputDirectory="$(dirname "${jspecify}")"
+  fi
 fi
-ourclasspath="${jspecify}:${dir}/build/libs/jspecify-reference-checker.jar"
+jspecifyReferenceChecker="${dir}/build/libs/jspecify-reference-checker.jar"
```
Comment (netdpb) [inline demo:18]: Shell style (at least hereabouts) has lower_camel_case for variables.  Can this be `jspecify_reference_checker`?

**Review by netdpb — APPROVED** [review]

File: demo (hunk 1 above)
Comment (netdpb) [reply demo:21]: Or I'll just do it.

File: demo (hunk 2 above)
Comment (netdpb) [reply demo:18]: I'll do it.

**Pull request merged**